src/svgp.jl

```diff
@@ -17,55 +20,60 @@ which can be found in closed form.
 variational Gaussian process classification." Artificial Intelligence and
 Statistics. PMLR, 2015.
 """
-function AbstractGPs.approx_posterior(::SVGP, fz::FiniteGP, q::AbstractMvNormal)
+function AbstractGPs.posterior(svgp::SVGP)
+    q, fz = svgp.q, svgp.fz
     m, A = mean(q), _chol_cov(q)
     Kuu = _chol_cov(fz)
     B = Kuu.L \ A.L
     α = Kuu \ (m - mean(fz))
-    data = (A=A, m=m, Kuu=Kuu, B=B, α=α, u=fz.x)
-    return ApproxPosteriorGP(SVGP(), fz.f, data)
+    data = (A=A, m=m, Kuu=Kuu, B=B, α=α)
+    return ApproxPosteriorGP(svgp, fz.f, data)
 end
 
-function Statistics.mean(f::ApproxPosteriorGP{SVGP}, x::AbstractVector)
-    return mean(f.prior, x) + cov(f.prior, x, f.data.u) * f.data.α
+AbstractGPs.posterior(svgp::SVGP, ::FiniteGP, ::AbstractVector) = posterior(svgp)
+
+function Statistics.mean(f::ApproxPosteriorGP{<:SVGP}, x::AbstractVector)
+    return mean(f.prior, x) + cov(f.prior, x, inducing_points(f)) * f.data.α
 end
 
-function Statistics.cov(f::ApproxPosteriorGP{SVGP}, x::AbstractVector)
-    Cux = cov(f.prior, f.data.u, x)
+function Statistics.cov(f::ApproxPosteriorGP{<:SVGP}, x::AbstractVector)
```

Comment:
Just for my own understanding: why would
```suggestion
function Statistics.cov(f::ApproxPosteriorGP{SVGP}, x::AbstractVector)
```
not work? Because SVGP is a parametric type dependent on Tfz and Tq?
Would 
```suggestion
function Statistics.cov(f::ApproxPosteriorGP{SVGP{Tfz,Tq}}, x::AbstractVector) where Tfz, Tq
```
work ?

Reply:
thanks:)
can you give any concrete examples where "spelling it out" would make a difference ?

Reply:
Also, what's the better style to use here - shorthand or longhand?

Reply:
just leave it as it is for now